plugins/samples/domain_rewrite/plugin.rs

```diff
@@ -0,0 +1,103 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START serviceextensions_plugin_domain_rewrite]
+use lol_html::*;
+use proxy_wasm::traits::*;
+use proxy_wasm::types::*;
+use std::cell::RefCell;
+use std::error::Error;
+use std::rc::Rc;
+
+proxy_wasm::main! {{
+    proxy_wasm::set_log_level(LogLevel::Trace);
+    proxy_wasm::set_http_context(|_, _| -> Box<dyn HttpContext> { MyHttpContext::new()});
+}}
+
+struct MyHttpContext<'a> {
+    // Some member variables need to be wrapped in Rc<Refcell<T>> so that we can
+    // modify the variable in multiple places as well as share ownership. By
+    // using Rc, we are commiting to not modfying the variable in multiple places
+    // simultaneously, otherwise the code will panic and plugin will crash.
+
+    // Stores HTML as rewriter parses and modifies HTML.
+    // Only stores completed sections(e.g., when rewriter parses "<di", output
+    // will be empty until the next chunk comes in. If the next chunk was
+    // "v> <h1>foo</h", the output will then contain "<div><h1>foo").
```

Comment:
Add a note here to explain that this only stores the HTML we haven't sent back to the client yet. As soon as we send it, we truncate the vector.